pandas/core/methods/selectn.py

```diff
@@ -111,13 +114,22 @@ def compute(self, method: str) -> Series:
         if n <= 0:
             return self.obj[[]]
 
-        dropped = self.obj.dropna()
-        nan_index = self.obj.drop(dropped.index)
+        # Save index and reset to default index to avoid performance impact
+        # from when index contains duplicates
+        original_index: Index = self.obj.index
+        cur_series = self.obj.reset_index(drop=True)
```

Comment:
What do you think of renaming `cur_series -> noindex` and `final_series -> result`

Reply:
How about `default_index` as it more accurately represents that the index is reset to default settings?

Reply:
Agreed - that's a better name.